Explicitly capture IxD specs and requirements within repo (#1128)
ni/nimble: specs/templates/component-interaction-design.md

# Interaction Design (IxD) [Template]

## Overview

*The name of the component, along with high-level description of what the design includes.*

### Background

*Relevant historical or background information*
*Link to Interaction Design issue*
*Link to relevant work items, related existing issues, etc.*

> ## Design

Author: Great point – I've added the following section…

--------

## Usage

*How, when, and where should you use this component?*
*Compare this component with other related components. How does it differ?*
*How should you decide between related components?*

> # Interaction Design (IxD) [Template]

Reviewer: Typically interaction designs are better captured in XD/Figma than in a markdown document. Should we update this template to make it clear that it's ok for the bulk of the design to live elsewhere? For example, change "Full description of the proposed interaction design" to "Link to design in XD or Figma and provide any additional context."

Do the other sections even provide value? For example, we don't usually see alternative designs called out in IxD specs. Maybe this whole template could go away and be replaced with a link from the IxD user story to the XD / Figma spec.

Author: > It probably doesn't need the "alternative implementations/designs" section.

Removed

>  For example, change "Full description of the proposed interaction design" to "Link to design in XD or Figma and provide any additional context."

Kept both

> ## Problem Statement

Reviewer 2: ```suggestion
## Overview
```

> *Brief description of the reason for this work and an executive summary of the proposed design.*

Reviewer 2: ```suggestion
*The name of the component, along with high-level description of what the design includes.*
```

> ## Links To Relevant Work Items and Reference Material

Reviewer 2: ```suggestion
### Background
```

*Include links to the work items this design addresses.*

> *Include links to any other meaningful reference material.*

Reviewer 2: ```suggestion
*Relevant historical or background information; link to UX issue implementation specification, etc.*
```

Author: I currently have the following, which works for me… let me know if you disagree.

```
*Relevant historical or background information*
*Link to Interaction Design issue*
*Link to relevant work items, related existing issues, etc.*
```

> ## Implementation / Design

Reviewer 2: ```suggestion
## Design
```

*Full description of the proposed interaction design.*

## Alternative Implementations / Designs

> *Describe any design alternatives and discuss why they were rejected.*

Reviewer 2: Removed (if this information is necessary, I will add it as needed)

## Open Issues

> *Describe any open issues with the design that you need feedback on before proceeding.*
> *It is expected that these issues will be resolved during the review process and this section will be removed when this documented in pulled into source.*

Reviewer 2: ```suggestion
*Highlight any open questions for discussion during the spec PR. Before the spec is approved these should typically be resolved with the answers being incorporated in the spec document.*
```
